### ISSUE TITLE
Use plugin cache dir with ephemeral agents

### ISSUE THREAD

**Issue by BenjaminDecreusefond** [AzBuilder/terrakube#1582]
### Feedback

Hi ! 

We tried to follow this [documentation](https://docs.terrakube.io/user-guide/workspaces/provider-cache) to use cache dir with ephemeral agents. Since this settings is made at the executor level we are not sure that effectively taken into account when a new ephemeral job is started. Beside that, using the option `TF_PLUGIN_CACHE_DIR=/home/cnb/.terraform.d/plugin-cache` in a workspace running on ephemeral agents throw error in plan `Error: The specified plugin cache dir /home/cnb/.terraform.d/plugin-cache/ cannot be opened: stat /home/cnb/.terraform.d/plugin-cache/: no such file or directory` even though after checking in the ephemeral job that the folder actually exists. Would you know what we've missed ?

Regards !

**Comment by BenjaminDecreusefond** [comment]
Yeah ! that's what I was thinking about :/ Thanks you ! I'll try to figure out how we can mitigate those storage issues !

**Comment by BenjaminDecreusefond** [comment]
Btw, I tested the toleration feature and it works good !

**Comment by alfespa17** [comment]
> Btw, I tested the toleration feature and it works good !

Cool, I was planing to release the new version tomorrow by the end of the day, let me know if you find any other issue or if you have any suggestion to add to the new version

**Comment by BenjaminDecreusefond** [comment]
I'm also curious to know. When we run an ephemeral agents, it uses the storage of the executor right ?  Would it make sense to create several executor to have different disk space ?

**Comment by alfespa17** [comment]
> I'm also curious to know. When we run an ephemeral agents, it uses the storage of the executor right ? Would it make sense to create several executor to have different disk space ?

It will use the container storage
